features/mbedtls/targets/TARGET_STM/sha256_alt.c

```diff
@@ -0,0 +1,141 @@
+{
+    if (ctx->is224 == 0) {
+        HAL_HASHEx_SHA256_Accumulate(&ctx->hhash_sha256, (uint8_t *) data, MBEDTLS_SHA256_BLOCK_SIZE);
+    } else {
+        HAL_HASHEx_SHA224_Accumulate(&ctx->hhash_sha256, (uint8_t *) data, MBEDTLS_SHA256_BLOCK_SIZE);
+    }
+}
+
+void mbedtls_sha256_update( mbedtls_sha256_context *ctx, const unsigned char *input, size_t ilen )
+{
+    size_t currentlen = ilen;
+    // store mechanism to handle MBEDTLS_SHA256_BLOCK_SIZE bytes per MBEDTLS_SHA256_BLOCK_SIZE bytes
+    if (currentlen == 0){ // only change HW status is size if 0
+        if(ctx->hhash_sha256.Phase == HAL_HASH_PHASE_READY) {
+            /* Select the SHA256 or SHA224 mode and reset the HASH processor core, so that the HASH will be ready to compute
+             the message digest of a new message */
+            if (ctx->is224 == 0) {
+                HASH->CR |= HASH_ALGOSELECTION_SHA256 | HASH_CR_INIT;
+            } else {
+                HASH->CR |= HASH_ALGOSELECTION_SHA224 | HASH_CR_INIT;
+            }
+        }
+        ctx->hhash_sha256.Phase = HAL_HASH_PHASE_PROCESS;
+    } else if (currentlen < (MBEDTLS_SHA256_BLOCK_SIZE - ctx->sbuf_len)) {
+        // only buffurize
+        memcpy(ctx->sbuf + ctx->sbuf_len, input, currentlen);
+        ctx->sbuf_len += currentlen;
+    } else {
+        // fill buffer and process it
+        memcpy(ctx->sbuf + ctx->sbuf_len, input, (MBEDTLS_SHA256_BLOCK_SIZE-ctx->sbuf_len));
+        currentlen -= (MBEDTLS_SHA256_BLOCK_SIZE - ctx->sbuf_len);
+        mbedtls_sha256_process(ctx, ctx->sbuf);
+        // now process every input as long as it is %4 bytes
+        size_t iter = currentlen / 4;
+        if (ctx->is224 == 0) {
+            HAL_HASHEx_SHA256_Accumulate(&ctx->hhash_sha256, (uint8_t *)(input + MBEDTLS_SHA256_BLOCK_SIZE - ctx->sbuf_len),  (iter * 4));
+        } else {
+            HAL_HASHEx_SHA224_Accumulate(&ctx->hhash_sha256, (uint8_t *)(input + MBEDTLS_SHA256_BLOCK_SIZE - ctx->sbuf_len),  (iter * 4));
+        }
+        // sbuf is now fully accumulated, now copy 1 / 2 or 3 remaining bytes
+        ctx->sbuf_len = currentlen % 4;
+        if (ctx->sbuf_len !=0) {
+            memcpy(ctx->sbuf, input + iter, ctx->sbuf_len);
+        }
+    }
+}
+
+void mbedtls_sha256_finish( mbedtls_sha256_context *ctx, unsigned char output[32] )
+{
+    if (ctx->sbuf_len > 0) {
+        if (ctx->is224 == 0) {
+            HAL_HASHEx_SHA256_Accumulate(&ctx->hhash_sha256, ctx->sbuf, ctx->sbuf_len);
+        } else {
+            HAL_HASHEx_SHA224_Accumulate(&ctx->hhash_sha256, ctx->sbuf, ctx->sbuf_len);
+        }
+    }
+    mbedtls_zeroize(ctx->sbuf, MBEDTLS_SHA256_BLOCK_SIZE);
+    ctx->sbuf_len = 0;
+    __HAL_HASH_START_DIGEST();
+        
+    if (ctx->is224 == 0)
+        HAL_HASHEx_SHA256_Finish(&ctx->hhash_sha256, output, 10);
+    else
+        HAL_HASHEx_SHA224_Finish(&ctx->hhash_sha256, output, 10);
```

Comment:
I am wondering whether this timeout value should be somehow configurable, perhaps through some `#define` or something...